### PR TITLE
Update density plots documentation

### PR DESCRIPTION
**Pull request by canyon289**
Addresses https://github.com/arviz-devs/arviz/issues/196

Showing one example for density plot. Let me know If this looks good and I can replicate across all other docs

### CODE REVIEW

**Comment by ColCarroll** [comment]
Do you think it is maintainable to try to keep that argument up-to-date, or should we say something like 

> refer to documentation of arviz.convert_to_dataset for details

**Comment by canyon289** [comment]
I think its better to refer to the inference_data documentation for us, but more annoying for the user.

But I agree with you and will make the change, it'll be harder to maintain this

**Comment by canyon289** [comment]
Updated per your comment, which basically means I copied and pasted what you wrote :)

Since we're doing this as part of this PR I need to expose the convert_to_inference_data documentation on the api docs

https://arviz-devs.github.io/arviz/api.html

**Comment by ColCarroll** [comment]
This is really great - will make library much friendlier to use. Feel free to merge whenever.

File: arviz/data/converters.py [hunk 1]
```diff
@@ -19,13 +18,13 @@ def convert_to_inference_data(obj, *, group='posterior', coords=None, dims=None,
...
         A supported object to convert to InferenceData:
-            InferenceData: returns unchanged
-            str: Attempts to load the netcdf dataset from disk
-            pystan fit: Automatically extracts data
-            pymc3 trace: Automatically extracts data
-            xarray.Dataset: adds to InferenceData as only group
-            dict: creates an xarray dataset as the only group
-            numpy array: creates an xarray dataset as the only group, gives the
+            | InferenceData: returns unchanged \n
+            | str: Attempts to load the netcdf dataset from disk \n
```
[the remark this answers is not in the document]
Comment (canyon289) [reply arviz/data/converters.py]: Ah whoops. This was me trying to get the linebreak to work but then forgetting

**Comment by canyon289** [comment]
Getting duplicate code errors from pylint. I'll merge after I figure out how to avoid the linting error

**Pull request merged**